src/app/directives/common/requires-roles/requires-roles.directive.ts

```diff
@@ -2,6 +2,7 @@ import {
   ComponentRef, Directive, HostBinding, Input, TemplateRef, ViewContainerRef,
 } from '@angular/core';
 import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import { isEqual } from 'lodash';
```

Comment:
Maybe it worth using `isMatch` here

Reply:
Would't `isMatch` perform a partial comparison or something like it?